Source/FieldSolver/SpectralSolver/SpectralBinomialFilter.cpp

```diff
@@ -0,0 +1,55 @@
+/* Copyright 2020 David Grote
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#include "SpectralBinomialFilter.H"
+#include <AMReX_REAL.H>
+
+#include <cmath>
+
+/* \brief Initialize the radial filter array */
+void
+SpectralBinomialFilter::InitFilterArrayR (HankelTransform::RealVector const & kr,
+                                          amrex::Real const dr,
+                                          int const npasses,
+                                          bool const compensation)
+{
+
+    // Note that this includes the kr values for all modes
+    filter_r.resize(kr.size());
+
+    for (int i=0 ; i < kr.size() ; i++) {
+        amrex::Real const sr = std::sin(0.5*kr[i]*dr);
+        amrex::Real const sr2 = sr*sr;
+        amrex::Real filt_r = std::pow(1. - sr2, npasses);
+        if (compensation) {
+            filt_r *= (1. + npasses*sr2);
+        }
```

Comment:
Would it make sense to add the `_rt` prefix for single precision runs?
```suggestion
        amrex::Real const sr = std::sin(0.5_rt*kr[i]*dr);
        amrex::Real const sr2 = sr*sr;
        amrex::Real filt_r = std::pow(1._rt - sr2, npasses);
        if (compensation) {
            filt_r *= (1._rt + npasses*sr2);
        }
```